source/extensions/common/aws/metadata_fetcher.cc

```diff
@@ -52,6 +52,12 @@ class MetadataFetcherImpl : public MetadataFetcher,
     ASSERT(!request_);
     complete_ = false;
     receiver_ = makeOptRef(receiver);
+
+    // Stop processing if we are shutting down
+    if (cm_.isShutdown()) {
```

Comment:
can we have a test for this?

Reply:
test case emulates shutdown, as well as the nullptr that would have crashed it had the fix not been in place